File: frontend/src/services/CompanyService.ts
import api from "./api";
import type { Company } from "../types/company";

export interface CompanyPage {
  content: Company[];
  totalPages: number;
  totalElements: number;
  number: number;
  size: number;
}

export const getCompanies = async (
  keyword: string,
  location: string,
  jobType: string,
  experience: string,
  page = 0,
  size = 10,
  sortBy = "postedDate",
  direction = "desc"
): Promise<CompanyPage> => {

  const response = await api.get("/companies/jobs", {
    params: {
      keyword,
      location,
      jobType,
      experience,
      page,
      size,
      sortBy,
      direction,
    },
  });

  return response.data.data;
};

export const getCompanyById = async (
  id: number
): Promise<Company> => {

  const response = await api.get(`/companies/${id}`);

  return response.data.data;

};

export const syncJobs = async (
  keyword: string
) => {

  const response = await api.post(
    `/companies/sync?keyword=${encodeURIComponent(keyword)}`
  );

  return response.data;

};

File: backend/src/main/java/com/interviewx/company/service/CompanyService.java
package com.interviewx.company.service;

import com.interviewx.company.dto.CompanyRequest;
import com.interviewx.company.dto.CompanyResponse;

import java.util.List;
import org.springframework.data.domain.Page;
;

public interface CompanyService {

    CompanyResponse createCompany(CompanyRequest request);

    List<CompanyResponse> getAllCompanies();

    CompanyResponse getCompanyById(Long id);

    List<CompanyResponse> searchCompanies(String keyword);
    

     int syncJobs(String keyword);
     Page<CompanyResponse> getJobs(

        String keyword,

        String location,

        String jobType,

        String experience,

        int page,

        int size,

        String sortBy,

        String direction

);

}